### ISSUE TITLE
On SmartTable, unselection doesn't call onSelectedItemsChange

### ISSUE THREAD

**Issue by Chinimala** [coreui/coreui-react#223]
On the SmartTable component, if I enable selectable and try to uncheck a line checkbox, the onSelectedItemsChange function doesn't always get called. So it's actually unusable with a react useState.

That's because, in the useEffect line 360 of CSmartTable.tsx :
```
    useEffect(() => {
      if (selectable) {
        const selected = _items.filter((item) => item._selected === true)

        if (selected.length === _items.length) {
          setSelectedAll(true)
          return
        }

        if (selected.length === 0) {
          setSelectedAll(false)
          return
        }

        if (selected.length !== 0 && selected.length !== _items.length) {
          setSelectedAll('indeterminate')
        }

        onSelectedItemsChange && onSelectedItemsChange(selected)
      }
    }, [_items])
```

You return in the first two if, so it doesn't call onSelectedItemsChange. Better use else if from the third case to the first :
```
    useEffect(() => {
      if (selectable) {
        const selected = _items.filter((item) => item._selected === true)

        if (selected.length !== 0 && selected.length !== _items.length) {
          setSelectedAll('indeterminate')
        } else if (selected.length === 0) {
          setSelectedAll(false)
        } else if (selected.length === _items.length) {
          setSelectedAll(true)
        }

        onSelectedItemsChange && onSelectedItemsChange(selected)
      }
    }, [_items])
```

**Comment by mrholek** [comment]
Fixed in v4.0.2

**Comment by Chinimala** [comment]
It works, thank you !
